ambry-router/src/test/java/com.github.ambry.router/GetManagerTest.java

```diff
@@ -133,6 +136,34 @@ public void testFailureOnAllPollThatSends()
       Assert.assertEquals("Exception received should be router closed error", RouterErrorCode.RouterClosed,
           routerException.getErrorCode());
     }
+
+    try {
+      router.getBlob(blobId).get();
+      Assert.fail("operation should have thrown");
+    } catch (ExecutionException e) {
+      RouterException routerException = (RouterException) e.getCause();
+      Assert.assertEquals("Exception received should be router closed error", RouterErrorCode.RouterClosed,
+          routerException.getErrorCode());
+    }
+  }
+
+  /**
+   * Do a getBlob on the given blob id and ensure that all the data is fetched and is correct.
+   * @param blobId the id of the blob (simple or composite) that needs to be fetched and compared.
+   * @throws Exception
+   */
+  private void getBlobAndCompareContent(String blobId)
+      throws Exception {
+    ByteBufferAsyncWritableChannel getChannel = new ByteBufferAsyncWritableChannel();
+    router.getBlob(blobId).get().readInto(getChannel, null);
+    int readBytes = 0;
+    do {
+      ByteBuffer buf = getChannel.getNextChunk();
+      while (buf.hasRemaining()) {
+        Assert.assertEquals("Get and Put blob content should match", putContent[readBytes++], buf.get());
```

Comment:
this will throw an `IndexOutOfBoundsException` if `putContent` runs out but `buf` has not. While you could argue that there is no problem since the test will fail and we will eventually debug, I would find it more informative if the test could check that the index has gone out of range and fail with a helpful message indicating the root cause i.e unexpected "extra" data returned.